packages/tuono-router/src/components/RouterContext.tsx

```diff
@@ -1,13 +1,29 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import type { ReactNode } from 'react'
 
 import type { Router } from '../router'
+import type { ServerRouterInfo } from '../types'
+
+export interface ParsedLocation {
+  href: string
+  pathname: string
+  search: Record<string, string>
+  searchStr: string
+  hash: string
+}
+
+export interface RouterContextType {
```

Comment:
```suggestion
export interface RouterContextValue {
```

I think we should changes this:
`RouterContextType` is an interface not a type.
I'm not a big fan of the [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation) like variables name